client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -670,18 +670,24 @@ function mapKeysRecursively( object, fn ) {
  *
  * @param {object} contactInformation - user's contact information
  * @param {string[]} domainNames - list of domain names
+ * @param {boolean | null} qualifyProperties If set and true, qualifies any nested property names in the returned error list
  * @param {Function} fn The callback function
  */
 Undocumented.prototype.validateDomainContactInformation = function(
 	contactInformation,
 	domainNames,
+	qualifyProperties,
```

Comment:
This works great! Still I have a suggestion: what if instead of adding a boolean in the middle of the argument list, we added an options object at the end of the argument list. Since arguments are always optional in JS, that would mean that existing calls to the function would not need to change (the options object would simply be undefined in those cases). New calls to the function could add `, { qualifyProperties: true } )` at the end of their calls, which also has the benefit of being more clear when reading than `, true )`. What do you think?

Reply:
Alternatively, we could change all the function arguments to properties of an object; that way they would be named and the order (or their absence) wouldn't matter. This would be more invasive, but equally useful.

So instead of

```js
function(
	contactInformation,
	domainNames,
	qualifyProperties,
        fn
)```

it would be

```js
function( {
	contactInformation,
	domainNames,
	qualifyProperties,
        fn
} )```

Reply:
An earlier version of this patch did add the parameter at the end, and I changed it because it felt unidiomatic for some reason. I do prefer the minimal nature of that, though, and using an object is a good idea. I'll make that edit.

Reply:
I agree about the idiomatic point; callbacks feel "right" at the end... but the challenge is that if we go down that road then either the callback should be converted to a Promise or some or all of the arguments should be put in an object, and this is the most minimal, but elegant, change I could think of.